vortex-expr/src/literal.rs

```diff
@@ -52,7 +56,14 @@ impl VortexExpr for Literal {
     }
 
     fn return_dtype(&self, _scope_dtype: &DType) -> VortexResult<DType> {
-        Ok(self.value.dtype().clone())
+        if self.value.is_valid() {
```

Comment:
Sorry, I missed this. Can you explain why you made this change? Not sure it's right per #2464 cc @AdamGS

Reply:
This fixed a test failure. But i think this is also incorrect. Maybe Ill see what failure this caused and fix it